Fix some errors in crm_wait_resource & pcs_property

furynick opened the pull request:
Fix #3
Fix #4

furynick committed:
Fix unknown states of promote resource

furynick committed:
Fix index out of range

Output of `pcs property config` with version 0.11, delimiter is still ":"
```
Cluster Properties:
 cluster-infrastructure: corosync
 cluster-name: d_artagnan
 dc-version: 2.1.5-a3f44794f94
 have-watchdog: false
```

OndrejHome commented:
Hello Nicola @furynick,

Thank you for including all proposed changes in the PR.

- as for the #3 as mentioned there I will need to have a better look at this as it seems on different systems I get different outputs.
- as for the #4 the PR would need to be opened against https://github.com/OndrejHome/ansible.crm_wait_for/pulls from where once merged I will get it here into `ansible_collection.ha_cluster`

Lets keep this PR around until we sort out the other ones and once those are done I will close it.

--
Ondrej

furynick commented:
PR for crm_wait_for is ready.

I guess there's some difference between different systems.
Perhaps the delimiter should be dynamically parsed ?

The idea should be to catch any "=" character in the output and if there's at least (output lines count) - 1 "=" that should be the delimiter. Otherwise it's ":"

It might be as simple as `delimiter = '=' if out.count('=') >= out.count('\n') - 2 else ':'`

I updated this PR accordingly and removed fix for crm_wait_for.

furynick committed:
Remove added states on this PR

furynick committed:
Parse delimiter automatically

furynick committed:
Improve version check

OndrejHome commented:
As the PR is related to changes to parts that are managed in separate repositories that were now merged, I will be closing this one without merge.
- https://github.com/OndrejHome/ansible.crm_wait_for/pull/1
- https://github.com/OndrejHome/ansible.pcs-modules-2/pull/51

All the changes are now part of ondrejhome.ha_cluster collection version `1.0.6`.
